Add preference menu for aug and sg

dcowen91 opened the pull request:
Implements the feature for issue #25.

This makes it so that players can opt in to receiving scoped rifles on CT & T side. If they opt in, then there is a 25% chance that they will receive an sg/aug instead of an m4/ak. I'm not married to this percentage at all if you have any feedback on how to better incorporate this feature.

dcowen91 committed:
add preference menu for aug and sg

splewis commented:
> If they opt in, then there is a 25% chance that they will receive an sg/aug instead of an m4/ak.

I think it'd be preferable to just have a generic rifle selection for both sides. CTs currently can pick silenced/unsilenced m4, the AUG should just be another entry.

dcowen91 commented:
> CTs currently can pick silenced/unsilenced m4, the AUG should just be another entry.

That makes sense. My only concern with letting every CT potentially have an AUG is that it might result in an unintended buff to CT sides, since the AUG makes certain angles a lot easier to hold. 

I'll go ahead and implement a generic rifle selection as you suggested and update this PR.

dcowen91 committed:
support rifle toggle menu (#1)

* update menu

* cleanup

* more cleanup

* rename

* fix tag

* implement weapon pref menu

dcowen91 commented:
I've implemented the rifle selection for CT and T sides. Do note however that I have not been able to test these changes- I'm primarily a Windows dev and there are some compatibility issues I'm facing with running smbuilder inside Windows Subsystem for Linux. I'll keep banging my head against those issues, but wanted to update the PR in the meantime.

splewis commented on scripting/include/executes.inc:
```diff
@@ -27,6 +27,17 @@ enum StratType {
+enum TRiflePref {
+  TRiflePref_Ak = 0,
+  TRiflePref_Sg = 1,
+}
```
Move these somewhere else, anything in executes.inc is intended to be part of the public API.

splewis commented on scripting/executes.sp:
```diff
@@ -1106,10 +1118,12 @@ public void UpdateTeams() {
         g_PlayerPrimary[client] = "weapon_ak47";
-      } else if (g_SilencedM4[client]) {
+      } else if (g_CTRifle[client] == CTRiflePref_M4) {
+        g_PlayerPrimary[client] = "weapon_m4a1";
```
I think this type of logic (same with the T side) should be encapsulated into another function `GivePlayerPrimaryRifle` or such. That way this logic doesn't need to be repeated so much.

splewis commented on scripting/executes.sp:
```diff
@@ -338,7 +337,6 @@ public void OnPluginStart() {
 
   g_AllowAWPCookie = RegClientCookie("executes_awpchoice", "", CookieAccess_Private);
-  g_SilencedM4Cookie = RegClientCookie("executes_silencedm4", "", CookieAccess_Private);
   g_CZCTSideCookie = RegClientCookie("executes_cz_ct_side", "", CookieAccess_Private);
```
I'm not sure what the implications of using cookies without ever registering them with `RegClientCookie` is. I'd rather continue to register the new ones as well, unless there's some documentation in SM around what happens here.

splewis commented:
> I've implemented the rifle selection for CT and T sides. Do note however that I have not been able to test these changes- I'm primarily a Windows dev and there are some compatibility issues I'm facing with running smbuilder inside Windows Subsystem for Linux. I'll keep banging my head against those issues, but wanted to update the PR in the meantime

Mostly looks good, a few notes.

You don't necessarily have to use smbuilder to compile this, you can compile it the 'standard way' for compiling sourcemod plugins (`spcomp executes.sp` or such). I mostly just use it to simplify packaging for servers & releases. (the travis.ci builds here are running the compilation too, so you *can* test it that way if you must :p)

dcowen91 committed:
update menu (#2)

* implement weapon pref menu

dcowen91 commented:
Thanks for the feedback! I addressed the comments and was finally able to test it! :)

![image](https://user-images.githubusercontent.com/6540263/62429205-74bc8a00-b6c0-11e9-8749-92aa01efa830.png)

dcowen91 committed:
remove dead code

splewis commented on scripting/executes/loadout_common.sp:
```diff
@@ -42,7 +42,24 @@ public char RandomNade() {
 }
 
-//
+public void GivePrimaryRifle(int client, int team) {
```
Rename this SetPrimaryRifle - GivePrimaryRifle makes me thing it's actually giving the weapon (like GivePlayerItem). Other than that everything looks good to me.

splewis commented:
Thanks for testing. One last nitpick.

dcowen91 committed:
rename
